agent/csi/volume_manager_test.go

```diff
@@ -0,0 +1,101 @@
+package csi
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/docker/swarmkit/agent/exec"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+const iterations = 25
+const interval = 100 * time.Millisecond
+
+func NewFakeManager() *volumes {
+	return &volumes{
+		m:         make(map[string]*api.VolumeAssignment),
+		pluginMap: make(map[string]*NodePlugin),
+	}
+}
+
+func TestTaskRestrictedVolumesProvider(t *testing.T) {
+	type testCase struct {
+		desc          string
+		volumeIDs     map[string]struct{}
+		volumes       exec.VolumeGetter
+		volumeID      string
+		taskID        string
+		volumeIDToGet string
+		value         string
+		expected      string
+		expectedErr   string
+	}
+
+	originalvolumeID := identity.NewID()
+	taskID := identity.NewID()
+	taskSpecificID := fmt.Sprintf("%s.%s", originalvolumeID, taskID)
```

Comment:
`taskSpecificID` is related to a feature of secrets, and does not need to be considered for volumes. And in either case, it seems like you don't have any code to handle it.